test/lit.cfg

```diff
@@ -417,6 +421,11 @@ if run_os.startswith('wasi'):
 # Parse the host triple
 (host_cpu, host_vendor, host_os, host_vers) = re.match('([^-]+)-([^-]+)-([^0-9-]+)(.*)', config.host_triple).groups()
 
+# Manually set the host OS on Android, because the host OS from the triple says
+# `linux`.
+if kIsAndroid:
+    host_os = 'android'
```

Comment:
Rather than overriding this just for the new `toolchain_lib_dir` below, I overrode `host_os` itself on Android, as it is only used for that and `target-same-as-host` just below, which is only checked by a single macOS-only test, so changing this for Android hosts should be fine.